File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/IntradayBarEvent.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/IntradayBarEvent.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Event;
import com.bloomberglp.blpapi.Message;
import com.bloomberglp.blpapi.Datetime;
import java.util.List;
import java.util.ArrayList;

public class IntradayBarEvent extends Event
{
	public IntradayBarEvent(IntradayBarRequest request) throws Exception
	{
		super._request = request;
		super._messages = this.generateMessages();
	}
	
    private List<Message> generateMessages() throws Exception
    {
        List<Message> result = new ArrayList<Message>();
        IntradayBarRequest ireq = (IntradayBarRequest)super._request;

        String security = ireq.security();
        boolean securityError = Rules.isSecurityError(security);
        if(securityError)
        {
            IntradayBarMessage msg = new IntradayBarMessage(super._request.correlationId(), ireq.getService(), security);
            result.add(msg);
        }
        else
        {
	        List<IntradayBarTickDataType> barData = new ArrayList<IntradayBarTickDataType>();	        
	        if(ireq.getDtStart() != null)
	        {
	        	List<Datetime> dates = ireq.getDateTimes();
	        	for(int i = 0; i < dates.size(); i++)
	        	{
	        		Datetime dtCurrent = dates.get(i);
	        		IntradayBarTickDataType value = com.bloomberglp.blpapi.RandomDataGenerator.generateBarData(dtCurrent);
	        		barData.add(value);
	        	}
	        }
	        
	        IntradayBarMessage msg = new IntradayBarMessage(super._request.correlationId(), barData, ireq.getService());
	        result.add(msg);
        }

        return result;
    }

    public List<Message> GetMessages()
    {
        return this._messages;
    }
	
}

File: BEmu/IntradayBarRequest/IntradayBarEvent.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="IntradayBarRequest/IntradayBarEvent.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi.IntradayBarRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class IntradayBarEvent : Event
    {
        private readonly List<Message> _messages;

        internal IntradayBarEvent(IntradayBarRequest request)
        {
            base._request = request;
            this._messages = this.GenerateMessages();
        }

        private List<Message> GenerateMessages()
        {
            List<Message> result = new List<Message>();
            var ireq = (IntradayBarRequest)base._request;

            string security = ireq.Security;
            bool securityError = Types.Rules.IsSecurityError(security);

            if (securityError)
            {
                IntradayBarMessage msg = new IntradayBarMessage(base._request.correlationId, ireq.Service, security);
                result.Add(msg);
            }
            else
            {
                var barData = new List<IntradayBarTickDataType>();
                if (ireq.DtStart.HasValue)
                {
                    foreach (var dtCurrent in ireq.GetDateTimes())
                    {
                        var value = Types.RandomDataGenerator.GenerateBarData(dtCurrent);
                        barData.Add(value);
                    }
                }
                IntradayBarMessage msg = new IntradayBarMessage(base._request.correlationId, ireq.Service, barData);
                result.Add(msg);
            }

            return result;
        }

        public override IEnumerable<Message> GetMessages()
        {
            return this._messages;
        }

    }
}
